relationalfield.go

```diff
@@ -121,6 +123,51 @@ func goDatatype(f *RelationalFieldDefinition, includePtr bool) string {
 	return "UNKNOWN TYPE"
 }
 
+func goDatatypeByModel(m *RelationalModelDefinition, belongsToModelName string) string {
+	f := m.RelationalFields[belongsToModelName+"ID"]
+	if f == nil {
+		return "int"
+	}
+	return belongsToIDType(f, true)
+}
+
+func belongsToIDType(f *RelationalFieldDefinition, includePtr bool) string {
+	if f.Parent == nil {
+		return "int"
+	}
+	modelName := strings.Replace(f.FieldName, "ID", "", -1)
+	model := f.Parent.BelongsTo[modelName]
```

Comment:
what happens when these two variables (modelName and model) come up empty?  Should there be error handling here?  same below.